client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -77,6 +81,8 @@ const PatternAssembler = ( {
 	const siteId = useSiteIdParam();
 	const siteSlugOrId = siteSlug ? siteSlug : siteId;
 	const locale = useLocale();
+	const isNewSite = !! useQuery().get( 'isNewSite' ) || isSiteSetupFlow( flow );
```

Comment:
This way to check if a site is new uses the entry points to the assembler are `with-theme-assembler` or `site-setup`. We can also use this way to identify new sites and add the site option `wpcom_site_setup=assembler`.

Another way is to use the property `time_posted` from the entry in `wp_signup_log`. We use it in the Blog Report Card to show how old a site is. We can assume that a site is new when it has been created in the past 24h.

What are you thoughts on using `time_posted`? It seems easier to maintain because the flows and assembler entry points will change in the future.

Reply:
We can get the creation time called `registered` from the `get_blog_details` function on the server site. If we can use the creation time to determine whether a site is newly created within the specified time range (e.g. 24 hours), then I can use that data 🙂

Let's create an issue and do it in the follow-up PR. WDTY?

Reply:
Oh, or we can simply use `site.options.created_at` to determine whether it's a new site or not

Reply:
But there might be some edge cases. For example, if the user creates a new site, and then goes through the `with-theme-assembler` flow from the Theme Showcase again within the specific time range, should we treat it a new site?

Reply:
Let's create an issue to discuss cons/pros and decide what solution is better and easier to maintain. I think there is not a perfect solution.

Reply:
What are your thoughts on this @taipeicoder?

Reply:
Here is the [isNewSite](https://github.com/Automattic/wp-calypso/blob/97a7f20365454dbb30c4b5fdd98e360a9ab93cb1/client/state/sites/selectors/is-new-site.js#L13) selector in Calypso now.